packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts

```diff
@@ -254,8 +254,8 @@ function mapDrawStyleToPathBuilder(
     builders = {
       linear: pathBuilders.linear!(),
       smooth: pathBuilders.spline!(),
-      stepBefore: pathBuilders.stepped!({ align: -1 }),
-      stepAfter: pathBuilders.stepped!({ align: 1 }),
+      stepBefore: pathBuilders.stepped!({ align: -1, extend: true }),
+      stepAfter: pathBuilders.stepped!({ align: 1, extend: true }),
```

Comment:
these bits will cause stepped interpolation to extend beyond the data to the edges of the plotting area:

![image](https://user-images.githubusercontent.com/43234/204402138-44e50599-5d7f-4389-8c3d-77a6584ff967.png)